client/src/actions/annotation.ts

```diff
@@ -0,0 +1,631 @@
+import { deflate } from "pako";
```

Comment:
Heads-up for reviewers: this new thunk module is the hub for all client-side persistence. In addition to the existing Redux actions it now wires in category/label CRUD, autosave batching (see `saveObsAnnotationsAction`), and the geneset PUT flow. Worth reading this before looking at the UI changes, since the components mostly dispatch these helpers.